vm/src/hint_processor/builtin_hint_processor/bootloader/simple_bootloader_hints.rs

```diff
@@ -0,0 +1,198 @@
+use crate::hint_processor::builtin_hint_processor::hint_utils::{
+    get_integer_from_var_name, get_ptr_from_var_name, insert_value_from_var_name,
+};
+use crate::hint_processor::hint_processor_definition::HintReference;
+use crate::serde::deserialize_program::ApTracking;
+use crate::types::errors::math_errors::MathError;
+use crate::types::exec_scope::ExecutionScopes;
+use crate::vm::errors::hint_errors::HintError;
+use crate::vm::vm_core::VirtualMachine;
+
+/// Implements
+/// n_tasks = len(simple_bootloader_input.tasks)
+/// memory[ids.output_ptr] = n_tasks
+///
+/// # Task range checks are located right after simple bootloader validation range checks, and
+/// # this is validated later in this function.
+/// ids.task_range_check_ptr = ids.range_check_ptr + ids.BuiltinData.SIZE * n_tasks
+///
+/// # A list of fact_toplogies that instruct how to generate the fact from the program output
+/// # for each task.
+/// fact_topologies = []
+pub fn prepare_task_range_checks(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    // n_tasks = len(simple_bootloader_input.tasks)
+    let simple_bootloader_input: SimpleBootloaderInput =
+        exec_scopes.get(vars::SIMPLE_BOOTLOADER_INPUT)?;
+    let n_tasks = simple_bootloader_input.tasks.len();
+
+    // memory[ids.output_ptr] = n_tasks
+    let output_ptr = get_ptr_from_var_name("output_ptr", vm, ids_data, ap_tracking)?;
+    vm.insert_value(output_ptr, Felt252::from(n_tasks))?;
+
+    // ids.task_range_check_ptr = ids.range_check_ptr + ids.BuiltinData.SIZE * n_tasks
+    const BUILTIN_DATA_SIZE: usize = 8;
+    let range_check_ptr = get_ptr_from_var_name("range_check_ptr", vm, ids_data, ap_tracking)?;
+    let task_range_check_ptr = (range_check_ptr + BUILTIN_DATA_SIZE * n_tasks)?;
+    insert_value_from_var_name(
+        "task_range_check_ptr",
+        task_range_check_ptr,
+        vm,
+        ids_data,
+        ap_tracking,
+    )?;
+
+    // fact_topologies = []
+    let fact_topologies = Vec::<FactTopology>::new();
+    exec_scopes.insert_value(vars::FACT_TOPOLOGIES, fact_topologies);
+
+    Ok(())
+}
+
+/// Implements
+/// from starkware.cairo.bootloaders.simple_bootloader.objects import Task
+///
+/// # Pass current task to execute_task.
+/// task_id = len(simple_bootloader_input.tasks) - ids.n_tasks
+/// task = simple_bootloader_input.tasks[task_id].load_task()
+pub fn set_current_task(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let simple_bootloader_input: SimpleBootloaderInput =
+        exec_scopes.get(vars::SIMPLE_BOOTLOADER_INPUT)?;
+    let n_tasks_felt =
+        get_integer_from_var_name("n_tasks", vm, ids_data, ap_tracking)?.into_owned();
+    let n_tasks = n_tasks_felt
+        .to_usize()
```

Comment:
A couple things scare me about this:
* `usize` is architecture dependent, and that should be an irrelevant detail here
* the intended bounds of this are unclear, although it's likely a small value
* it's worth auditing that this isn't user influenceable to the extent that a malicious user could cause an overflow, etc.

Reply:
As with my comment on another PR, it might be safe to just use some implementation of `U256` here.

Reply:
I used `usize` because we use this value later to index a vector. IMO this is safe as we just return an error if `n_tasks` is larger than `usize`.

Reply:
I still think the outcome shouldn't depend no the architecture. Maybe we should use the smaller of expected `usize` values to make sure the behavior is consistent.

Reply:
We'd need to cast to `usize` anyway as it's used a vector index. Otherwise the program simply does not compile.

Note that `usize` is [guaranteed to be 8 bytes on 64-bit machines](https://doc.rust-lang.org/std/primitive.usize.html).

Overall I just don't see a security risk here. Sure, someone can compile this as a 32-bit executable, but then they'll just get an error when running the prover API on their node if `n_tasks > UINT_MAX`. `n_tasks` should be equal to the number transactions to prove, so this is not a parameter directly controlled by users.

Reply:
I'll drop the old bootloader code for context here: https://github.com/lambdaclass/cairo-lang/blob/02d511208cece64da1300445ab0586cf75205e70/src/starkware/cairo/bootloaders/simple_bootloader/run_simple_bootloader.cairo#L153

I agree that it's not really clear what the hint function you've written could do to improve at the moment.